activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -248,11 +248,23 @@ def add_column(table_name, column_name, type, **options) #:nodoc:
       end
 
       def remove_column(table_name, column_name, type = nil, **options) #:nodoc:
-        alter_table(table_name) do |definition|
-          definition.remove_column column_name
-          definition.foreign_keys.delete_if do |_, fk_options|
-            fk_options[:column] == column_name.to_s
+        old_way = ->() {
+          alter_table(table_name) do |definition|
+            definition.remove_column column_name
+            definition.foreign_keys.delete_if do |_, fk_options|
+              fk_options[:column] == column_name.to_s
+            end
           end
+        }
+
+        if ::SQLite3::SQLITE_VERSION_NUMBER >= 3035000
+          begin
+            super
+          rescue ActiveRecord::StatementInvalid => e
```

Comment:
This handles cases that were working before but now fails because of the native drop column limitations: https://sqlite.org/lang_altertable.html#altertabdropcol